projector/binding.go

```diff
@@ -415,3 +421,23 @@ func (p *serviceBindingProjector) providerAnnotation(binding *servicebindingv1be
 func (p *serviceBindingProjector) providerAnnotationName(binding *servicebindingv1beta1.ServiceBinding) string {
 	return fmt.Sprintf("%s%s", ProviderAnnotationPrefix, binding.UID)
 }
+
+func (p *serviceBindingProjector) bindingSecrets(binding *servicebindingv1beta1.ServiceBinding, workload runtime.Object) (string, error) {
+	if secret := p.secretName(binding); secret != "" {
+		return secret, nil
+	}
+	data, err := runtime.DefaultUnstructuredConverter.ToUnstructured(workload)
+	if err != nil {
+		return "", err
+	}
+	secret := corev1.Secret{}
```

Comment:
why would the workload be a secret?

Reply:
Because this code is definitely wrong.  Fixed.